src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -45,6 +50,142 @@ struct ProblemDescription;
 
 namespace solver {
 #if MIOPEN_BACKEND_HIP && MIOPEN_USE_COMPOSABLEKERNEL
+namespace batchnorm {
```

Comment:
this code should be moved back to a header used by `batchnorm` solvers only if there is common code between forward and backward batchnorm CKArgs etc., then create a separate header otherwise, move back to cpp files. 

Structure the `CKArgs` class similar to how convolution's CKArgs classes are structured where there's a method called `MakeArumentPointer`.

Reply:
@amberhassaan ,  I have reverted the changes in implicitgemm_ck_util.hpp. Also refactored BnCKFwdInference::GetSolution

src/solver/batchnorm/backward_ck.cpp
src/solver/conv/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp